ovh/resource_iam_resource_tags.go

```diff
@@ -0,0 +1,218 @@
+				Detail:   fmt.Sprintf("Tag value '%s' for key '%s' must match pattern ^[a-zA-Z0-9_.:/=+@-]{0,256}$", value, key),
+			})
+		}
+	}
+
+	return diags
+}
+
+// suppressOvhTagsDiff suppresses diffs for ovh: prefixed tags
+// These tags are managed by OVH and should not cause drift detection
+func suppressOvhTagsDiff(k, old, new string, d *schema.ResourceData) bool {
+	// k will be in format "tags.key_name" or just "tags" for the whole map
+	// We only want to suppress individual tag keys that start with "ovh:"
+	if strings.HasPrefix(k, "tags.") {
+		tagKey := strings.TrimPrefix(k, "tags.")
+		// Suppress diff if this is an ovh: prefixed tag
+		if strings.HasPrefix(tagKey, "ovh:") {
+			return true
+		}
+	}
+	return false
+}
+
+func resourceIamResourceTagsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+	resourceURN := d.Get("resource_urn").(string)
+	endpoint := fmt.Sprintf("/v2/iam/resource/%s", url.PathEscape(resourceURN))
+
+	// Get tags from configuration and convert to map[string]string
+	configTags := d.Get("tags").(map[string]interface{})
+	tagsMap := make(map[string]string)
+	for key, val := range configTags {
+		tagsMap[key] = val.(string)
+	}
+
+	// Update resource with tags using PUT endpoint
+	// The API will preserve ovh: prefixed system tags automatically
+	updateBody := map[string]interface{}{
+		"tags": tagsMap,
+	}
+
+	err := config.OVHClient.PutWithContext(ctx, endpoint, updateBody, nil)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to create tags: %w", err))
+	}
+
+	// Set the resource ID to the URN
+	d.SetId(resourceURN)
+
+	// Read back the resource to sync state
+	return resourceIamResourceTagsRead(ctx, d, meta)
+}
+
+func resourceIamResourceTagsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+	resourceURN := d.Id()
+
+	// Fetch the resource details
+	var resourceDetails IamResourceDetails
+	endpoint := fmt.Sprintf("/v2/iam/resource/%s", url.PathEscape(resourceURN))
+	err := config.OVHClient.GetWithContext(ctx, endpoint, &resourceDetails)
+	if err != nil {
+		return diag.Errorf("Error calling GET %s:\n\t %q", endpoint, err)
+	}
+
+	// Sync all tags to state (including ovh: prefixed system tags)
+	// This allows users to see all tags, but ovh: prefixed tags cannot be modified
+	allTags := make(map[string]string)
+	for key, value := range resourceDetails.Tags {
+		allTags[key] = value
+	}
+
+	if err := d.Set("tags", allTags); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceIamResourceTagsImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	// The ID passed to import is the resource URN
+	resourceURN := d.Id()
+
+	// Set the resource_urn field from the import ID
+	if err := d.Set("resource_urn", resourceURN); err != nil {
+		return nil, err
+	}
+
+	// Call Read to populate the rest of the state
```

Comment:
calling the Read function is not needed since Terraform already calls it after running the import func (the import is just supposed to set the right fields so that the read can be done) ;) In your case, the read will be called two times.